### PR TITLE
Add Docblocks

### PR DESCRIPTION
**Pull request by mikeselander**
As promised, here's a mostly-comprehensive PR adding basic docblocks to functions. This should most definitely be reviewed for accuracy as I am making some educated guesses here.

I left the hooks file alone as I already spent a bit too much time on this.

### CODE REVIEW

**Comment by mikeselander** [comment]
@rmccue would you mind glancing through this and verifying the accuracy of these comments?

File: src/diff.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
 const parseDiff = require( 'parse-diff' );
 
+/**
+ * Fetch which files are different in a particular push changeset.
```
Comment (rmccue) [inline src/diff.js:4]: This is actually about mapping the lines in our files to lines in the diff; we used to use it for the inline comments before GH checks existed, and it's what generates the "x errors, but none on lines in this PR" (which is broken)

```suggestion
 * Work out which files and lines are contained in this PR.
```

File: src/format.js [hunk 2]
```diff
@@ -19,6 +33,12 @@ const formatSummary = status => {
 	return summaryBits.join( ', ' );
 };
 
+/**
+ * Format a string that compares a previous an current code changesets.
```
Comment (rmccue) [inline src/format.js:37]: This compares the previous lint result against the current one to work out which errors have been fixed; I actually don't think this is used any more maybe?

```suggestion
 * Format a string that compares previous and current check runs.
```

File: src/format.js [hunk 3]
```diff
@@ -48,8 +68,14 @@ const formatComparison = comparison => {
 		.join( '\n' );
 
 	return status;
-}
+};
 
+/**
+ * Fetch linting results broken out by filename.
```
Comment (rmccue) [inline src/format.js:74]:
```suggestion
 * Group linting results by filename.
```

File: src/format.js [hunk 4]
```diff
@@ -205,16 +265,29 @@ const formatMetadata = context => {
 	return body;
 };
 
+/**
+ * Format a welcome message for users who have just activated the bot on the repo.
+ *
+ * @param {Object} state   Results of a linting run.
+ * @param {String} gistUrl URL of the gist where the cumulative initial results are stored.
+ * @returns {string} Welcome data.
+ */
```
Comment (rmccue) [inline src/format.js:274]: Sidenote: this is totally broken right now.

File: src/format.js [hunk 5]
```diff
@@ -205,16 +265,29 @@ const formatMetadata = context => {
...
 	body += ` I found ${ formatSummary( state ) } in your project.`;
-	body += `\n\nFor more information about hm-linter, see [the project repo](https://github.com/humanmade/linter-bot).`
+	body += `\n\nFor more information about hm-linter, see [the project repo](https://github.com/humanmade/linter-bot).`;
 	body += ` If you need a hand with anything, ping @rmccue or @joehoyle who are always happy to help.`;
 	body += `\n\n:heart: :robot:`;
 	return body;
 };
 
+/**
+ * Convert the whole results of a linting run into a JSON string.
```
Comment (rmccue) [inline src/format.js:286]:
```suggestion
 * Convert the whole results of a linting run into a JSON string for debugging.
```

File: src/linters/eslint/index.js [hunk 6]
```diff
@@ -3,6 +3,12 @@ const Module = require( 'module' );
 const path = require( 'path' );
 const moduleAlias = require( 'module-alias' );
 
+/**
+ * Format message data into a consistent format for usage in formatOutput.
```
Comment (rmccue) [inline src/linters/eslint/index.js:7]:
```suggestion
 * Convert a ESLint error into formatOutput-style results.
```

File: src/linters/eslint/index.js [hunk 7]
```diff
@@ -13,6 +19,13 @@ const formatMessage = message => {
 	};
 };
 
+/**
+ * Organize all output from eslint for Linter ingestion.
```
Comment (rmccue) [inline src/linters/eslint/index.js:23]:
```suggestion
 * Convert ESLint results into common output format.
```

File: src/linters/eslint/index.js [hunk 8]
```diff
@@ -63,8 +76,21 @@ const run = ( engine, codepath ) => {
...
+ *
+ * @type {string}
+ */
 const DEFAULT_STANDARD = process.env.DEFAULT_STANDARD_ESLINT || 'eslint-config-humanmade';
 
+/**
+ * Run eslint linting.
+ *
+ * @param {String} standardPath Path to custom standard set.
+ * @returns {function(*=): Promise<any>}
```
Comment (rmccue) [inline src/linters/eslint/index.js:92]: wat

File: src/linters/index.js [hunk 9]
```diff
@@ -12,6 +12,12 @@ const enabled = ( process.env.ENABLED_LINTERS || 'eslint,phpcs' ).split( ',' );
 const STANDARDS_DIR = '/tmp/hmlinter-standards';
 const BASE_URL = process.env.STANDARD_URL || 'https://make.hmn.md/hmlinter/standards';
 
+/**
+ * Fetch a file with a given path.
```
Comment (rmccue) [inline src/linters/index.js:16]:
```suggestion
 * Send a HTTP request.
 *
 * Promisified version of Node's https.get
```

File: src/linters/index.js [hunk 10]
```diff
@@ -40,6 +53,13 @@ const downloadFile = async ( url, filename ) => {
 	return await probotUtil.file.saveDownloadedFile( res.body, filename );
 };
 
+/**
+ * Prepare a linter for use in parsing files.
```
Comment (rmccue) [inline src/linters/index.js:57]:
```suggestion
 * Download and build a linter instance.
```

File: src/linters/phpcs/index.js [hunk 11]
```diff
@@ -10,6 +10,12 @@ const CONFIG_NAMES = [
 	'phpcs.ruleset.xml',
 ];
 
+/**
+ * Format message data into a consistent format for usage in formatOutput.
```
Comment (rmccue) [inline src/linters/phpcs/index.js:14]:
```suggestion
 * Convert a phpcs error into formatOutput-style results.
```

File: src/linters/phpcs/index.js [hunk 12]
```diff
@@ -23,6 +29,13 @@ const formatMessage = message => {
 	};
 };
 
+/**
+ * Organize all output from PHPCS for Linter ingestion.
```
Comment (rmccue) [inline src/linters/phpcs/index.js:33]:
```suggestion
 * Convert phpcs results into common output format.
```

File: src/metadata.js [hunk 13]
```diff
@@ -1,5 +1,16 @@
+/**
+ * Regex pattern for evaluating HM Linter comments and pulling a JSON blob out.
+ *
+ * @type {RegExp}
+ */
 const regex = /\n\n<!-- hm-linter = (.*) -->/;
 
+/**
+ * Convert an HM Linter HTML comment into a JSON object.
```
Comment (rmccue) [inline src/metadata.js:9]:
```suggestion
 * Parse HM Linter data from HTML comment text.
```

File: src/util.js [hunk 14]
```diff
@@ -5,6 +5,12 @@ const githubApi = require( '@octokit/rest' );
 
 const GIST_ACCESS_TOKEN = process.env.GIST_ACCESS_TOKEN || null;
 
+/**
+ * Assemble all linter results into an object broken out by linter type.
```
Comment (rmccue) [inline src/util.js:9]:
```suggestion
 * Combine results-by-linter into a single results object.
```

File: src/util.js [hunk 15]
```diff
@@ -20,6 +26,14 @@ function combineLinters( results ) {
 		}, {} );
 }
 
+/**
+ * Create a GitHub Gist for displaying messages in the gitHub UI.
```
Comment (rmccue) [inline src/util.js:30]: The function's a bit more generic:

```suggestion
 * Create a GitHub Gist (used for logging raw lint data).
```

File: src/linters/eslint/index.js (hunk 8 above)
Comment (mikeselander) [reply src/linters/eslint/index.js:92]: Oh, PHPStorm autofill nonsense

**Comment by mikeselander** [comment]
Ready or you to look at one more time @rmccue

**Pull request merged**